tests/tests/deployment.rs

```diff
@@ -2,23 +2,27 @@ use fuel_core::chain_config::ChainConfig;
 use fuel_core_types::fuel_tx::GasCosts;
 
 #[test_case::test_case(
-    include_str!("../../deployment/scripts/chainspec/beta_chainspec.json"); 
+    "./../deployment/scripts/chainspec/beta";
     "Beta chainconfig"
 )]
 #[test_case::test_case(
-    include_str!("../../deployment/scripts/chainspec/dev_chainspec.json"); 
-    "Dev chainconfig"
+    "./../deployment/scripts/chainspec/dev";
+    "Dev chainconfig" 
 )]
-fn test_deployment_chainconfig(bytes: &str) {
-    let mut chain_config: ChainConfig =
-        serde_json::from_str(bytes).expect("Should be able to decode chain config");
+fn test_deployment_chainconfig(path: &str) {
+    let chain_config = ChainConfig::load_from_directory(path)
+        .expect("Should be able to load chain config");
+
+    // TODO: add state config check in subsequent PR
+
     let benchmark_gas_costs =
         GasCosts::new(fuel_core_benches::default_gas_costs::default_gas_costs());
+
     // Deployment configuration should use gas costs from benchmarks.
-    chain_config.consensus_parameters.gas_costs = benchmark_gas_costs;
-    let actual_bytes = serde_json::to_string_pretty(&chain_config)
-        .expect("Should be able to encode the chain config");
-    assert_eq!(actual_bytes, bytes);
+    assert_eq!(
+        chain_config.consensus_parameters.gas_costs,
+        benchmark_gas_costs
+    );
```

Comment:
Setting the `gas_costs = benchmark_gas_costs;` enforced the prices according to latest benchmarks. Could we return it back?=)

Reply:
I'm not sure I follow what the issue is:

Previously the test had:
1. loaded the configuration from the file
2. changed the gas costs of the *loaded* configuration
3. checked that the loaded configuration, when serialized, is equal to the one from the file.

Had the gas costs in the file been anything other than what was checked in step #3 then the test would have failed.

So we interpreted the test as to mean:

Validate that:
1. you can decode the config (i.e. given file is a valid configuration)
2. check that the gas costs encoded in the file are equal to the given value (`benchmark_gas_costs`)

Since we needed to touch these tests for our changes we changed the assertion to be more direct. 
1. Check that the configuration is valid (i.e. can be decoded)
2. Check that the gas costs loaded from the file are as we expect.

We didn't check the encoding again since that is a general test that would belong over where the rest of the encoding/decoding is being tested -- i.e. the goal here is not to test the encoding/decoding but rather test that the committed configuration is valid and has expected values (i.e. prevent anyone from changing the gas costs accidentally)

Did we understand this right?

Reply:
Initially, this test was added to verify the decoding and encoding of the chain configuration. At the same time, enforce the same layout inside the chain configuration as during encoding.

It is also our way to update chain configs by copying/pasting the result from the test into chain configuration files directly.

Later, I also added a check for gas costs later as one of the requirements for these configurations.

Without testing encoding, we can't be sure that it is exactly the same config. Plus, removing it makes the process of updating `beta_chainspec.json` and `dev_chainspec.json` much harder=)